smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -281,4 +305,43 @@ public Void serviceShape(ServiceShape shape) {
 
         return null;
     }
+
+    private boolean shapesMatch(Shape shape, Shape otherShape) {
+        // Check names match.
+        if (!shape.getId().getName().equals(otherShape.getId().getName())) {
+            return false;
+        }
+
+        // Check traits match.
+        Map<ShapeId, Trait> traits = shape.getAllTraits();
+        Map<ShapeId, Trait> otherTraits = otherShape.getAllTraits();
+        for (ShapeId trait : traits.keySet()) {
+            // Ignore box trait when comparing.
```

Comment:
Why ignore the box trait?

Reply:
The box trait needs ignored for examples like this:
https://github.com/aws/aws-sdk-js-v3/blob/smithy-codegen/models/iot.2015-05-28.json#L10612:L10620

Those would generate the same TypeScript, so we don't want to throw that they're colliding shapes in this instance.  Since the box trait only applies to primitives, there won't have an issue with duplicate model code being generated.

Reply:
:+1: Can this be simplified to:

```java
Map<ShapeId, Trait> traitsA = new HashMap<>(shape.getAllTraits());
Map<ShapeId, Trait> traitsB = new HashMap<>(shape.getAllTraits());
// Ignore the box trait since it has no effect in JavaScript.
traitsA.remove(BoxTrait.ID);
traitsB.remove(BoxTrait.ID);
if (!traitsA.equals(traitsB)) {
    return false;
}
```

Reply:
Fixed in 66e0ca0